### PR TITLE
decode batch message

### PR DESCRIPTION
**Pull request by LvBay**
Provides a method for parsing the payload generated by the producer using the batch produce option.

### CODE REVIEW

File: core/msg/message_test.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package msg
+
+import(
+	"testing"
+)
+
+func TestDecodeBatchPayload(t *testing.T){
+	payload := []byte{0,0,0,2,24,12,104,101,108,108,111,45,112,117,108,115,97,114} // hello-pulsar
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
	payload := []byte{0, 0, 0, 2, 24, 12, 104, 101, 108, 108, 111, 45, 112, 117, 108, 115, 97, 114} // hello-pulsar
```

File: core/msg/message_test.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	list,err := DecodeBatchPayload(payload,1)
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
	list, err := DecodeBatchPayload(payload, 1)
```

File: core/msg/message_test.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	if err!=nil{
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
	if err != nil {
```

File: core/msg/message_test.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+		t.Fatal(err)
+	}
+	if get,want := len(list),1;get!=want{
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
	if get, want := len(list), 1; get!=want {
```

File: core/msg/message_test.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+		t.Errorf("want %v, but get %v",get,want)
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
		t.Errorf("want %v, but get %v", get, want)
```

File: core/msg/message_test.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	}
+	
+	m := list[0]
+	if get,want := string(m.SinglePayload), "hello-pulsar";get!=want{
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
	if get, want := string(m.SinglePayload), "hello-pulsar"; get != want {
```

File: core/msg/message_test.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+		t.Errorf("want %v, but get %v",get,want)
```
Comment (wolfstudy) [inline core/msg/message_test.go]:
```suggestion
		t.Errorf("want %v, but get %v",get, want)
```

File: core/msg/message.go [hunk 8]
```diff
@@ -38,3 +41,54 @@ func (m *Message) Equal(other *Message) bool {
 		proto.Equal(m.Meta, other.Meta) &&
 		bytes.Equal(m.Payload, other.Payload)
 }
+
+type SingleMessage struct {
```
Comment (cch123) [inline core/msg/message.go]: exported struct should have comment

File: core/msg/message.go [hunk 9]
```diff
@@ -38,3 +41,54 @@ func (m *Message) Equal(other *Message) bool {
...
+	SingleMetaSize uint32
+	SingleMeta     *api.SingleMessageMetadata
+	SinglePayload  []byte
+}
+
+func DecodeBatchMessage(msg *Message)([]*SingleMessage, error){
```
Comment (cch123) [inline core/msg/message.go]: exported function should have comment

**Comment by LvBay** [comment]
> nice job, left a little comments, can you fix it?

Sorry for so many problems of format, i have fix them. 

And thank you very much for pointing out every mistake.

@wolfstudy @cch123

**Pull request merged**